### PR TITLE
Adding bm60ec rgb

### PR DESCRIPTION
**Pull request by umbacos**
Hi, I wrote the keyboard "bm60ec rgb" which is the bm60 with rotary encoder that was missing in the QMK master branch.

## Description

created a new directory under keyboard called bm60ec_rgb, and added all the required files, including via support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Please revert all changes outside of the keyboard addition, itself.

File: .gitignore [hunk 1]
```diff
@@ -80,3 +80,7 @@ __pycache__
 
 # Allow to exist but don't include it in the repo
 user_song_list.h
+lib/ugfx
+lib/chibios
+lib/chibios
+lib/ugfx
```
Comment (drashna) [inline .gitignore:86]: Please revert this change.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
You have a lot of additions changes outside of the bm60ec rgb keyboard.  

Please revert these changes. If you need help with this, don't hesitate to ask.

File: keyboards/40percentclub/gherkin/keymaps/umbacos/config.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#pragma once
```
Comment (drashna) [inline keyboards/40percentclub/gherkin/keymaps/umbacos/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/40percentclub/gherkin/keymaps/umbacos/config.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#define MATRIX_COL_PINS { D0, D4, C6, D7, E6, B4 }
+*/
+
+#define IGNORE_MOD_TAP_INTERRUPT
+
+#define MUSIC_MAP
+
+#define RGB_DI_PIN D3
+#define RGBLED_NUM 20
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/40percentclub/gherkin/keymaps/umbacos/config.h:16]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/40percentclub/gherkin/keymaps/umbacos/keymap.c [hunk 4]
```diff
@@ -0,0 +1,252 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/40percentclub/gherkin/keymaps/umbacos/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/40percentclub/gherkin/keymaps/umbacos/keymap.c [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+#include "keymap_italian.h"
+#include <print.h>
+
+#ifdef AUDIO_ENABLE
+  #include "audio.h"
+#endif
```
Comment (drashna) [inline keyboards/40percentclub/gherkin/keymaps/umbacos/keymap.c:7]:
```suggestion
```

File: keyboards/40percentclub/gherkin/keymaps/umbacos/keymap.c [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/40percentclub/gherkin/keymaps/umbacos/keymap.c:9]:
```suggestion
```

File: keyboards/40percentclub/gherkin/keymaps/umbacos_light/config.h [hunk 7]
```diff
@@ -0,0 +1,25 @@
+#pragma once
```
Comment (drashna) [inline keyboards/40percentclub/gherkin/keymaps/umbacos_light/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/40percentclub/gherkin/keymaps/umbacos_light/config.h [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+#undef MATRIX_COL_PINS
+#define MATRIX_ROW_PINS { B6, B2, B3, B1, F7 }
+#define MATRIX_COL_PINS { D0, D4, C6, D7, E6, B4 }
+*/
+
+#define IGNORE_MOD_TAP_INTERRUPT
+
+#define RGB_DI_PIN F6
+#define RGBLED_NUM 6
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/40percentclub/gherkin/keymaps/umbacos_light/config.h:14]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/40percentclub/gherkin/keymaps/umbacos_light/keymap.c [hunk 9]
```diff
@@ -0,0 +1,80 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/40percentclub/gherkin/keymaps/umbacos_light/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/40percentclub/gherkin/keymaps/umbacos_midi/config.h [hunk 10]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (drashna) [inline keyboards/40percentclub/gherkin/keymaps/umbacos_midi/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by filterpaper** [comment]
KPrepublic PCBs are now located within `keyboards/kprepublic`. `bm60ec_rgb` should reside in there as well.

**Comment by peepeetee** [comment]
Also, I standardized the naming format for KPrepublic PCBs. (see #15047) See if you can rename the board to bm60ec

**Pull request closed without merging**